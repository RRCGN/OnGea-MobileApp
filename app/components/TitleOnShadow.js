/**
 * Title on a Inset Shadow.
 * Has a big Title and maybe a smaller Subtitle.
 * @flow
 */

import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'


type Props = {
  title: string,
  subtitle?: ?string
}

const TitleOnShadow = ({ title, subtitle }: Props) => (
  <LinearGradient
    colors={['rgba(0,0,0,0)', 'rgba(0,0,0,0.8)']}
    style={styles.overlay}
  >
    <Text style={[styles.text, styles.title]}>{title}</Text>
    {subtitle && <Text style={[styles.text, styles.subtitle]}>{subtitle}</Text>}
  </LinearGradient>
)

export default TitleOnShadow


const styles = StyleSheet.create({
  overlay: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    paddingTop: 24,
    paddingBottom: 16,
    paddingLeft: 16,
    paddingRight: 16
  },
  text: {
    color: 'white',
    backgroundColor: 'transparent'
  },
  title: {
    fontSize: 24
  },
  subtitle: {
    fontSize: 14
  },
})
