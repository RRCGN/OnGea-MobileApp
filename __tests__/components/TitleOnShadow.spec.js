import 'react-native'
import React from 'react'
import TitleOnShadow from '../../app/components/TitleOnShadow'

import renderer from 'react-test-renderer'

describe('<TitleOnShadow />', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <TitleOnShadow title="Foo" subtitle="bar" />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly without subtitle', () => {
    const tree = renderer.create(
      <TitleOnShadow title="Foo" />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
