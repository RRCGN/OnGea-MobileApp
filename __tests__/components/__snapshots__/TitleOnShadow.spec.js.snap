exports[`<TitleOnShadow /> renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      0,
      3422552064,
    ]
  }
  end={undefined}
  locations={null}
  start={undefined}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 24,
      "position": "absolute",
      "right": 0,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgba(255, 255, 255, 1)",
        },
        Object {
          "fontSize": 24,
        },
      ]
    }>
    Foo
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgba(255, 255, 255, 1)",
        },
        Object {
          "fontSize": 14,
        },
      ]
    }>
    bar
  </Text>
</BVLinearGradient>
`;

exports[`<TitleOnShadow /> renders correctly without subtitle 1`] = `
<BVLinearGradient
  colors={
    Array [
      0,
      3422552064,
    ]
  }
  end={undefined}
  locations={null}
  start={undefined}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 24,
      "position": "absolute",
      "right": 0,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgba(255, 255, 255, 1)",
        },
        Object {
          "fontSize": 24,
        },
      ]
    }>
    Foo
  </Text>
</BVLinearGradient>
`;
